internal/jimm/jujuauth/migrationtokens.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Canonical.
+
+package jujuauth
+
+import (
+	"context"
+	"time"
+
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/names/v5"
+
+	"github.com/canonical/jimm/v3/internal/jimmjwx"
+)
+
+const (
+	migrationExpiry = 3 * time.Hour
+)
+
+// MigrationTokenGenerator is responsible for generating migration tokens.
+// It uses the JWTService to create a token with specific claims for migration purposes.
+// The token includes the user, model tag, JIMM's UUID and an indication that it is for migration.
+type MigrationTokenGenerator struct {
+	jwtService JWTService
+	jimmUUID   string
+}
+
+// newMigrationTokenGenerator creates a new MigrationTokenGenerator instance.
+// It requires a JWTService to generate the tokens and JIMM's UUID used for the audience claim.
+func newMigrationTokenGenerator(jwtService JWTService, jimmUUID string) MigrationTokenGenerator {
+	return MigrationTokenGenerator{
+		jwtService: jwtService,
+		jimmUUID:   jimmUUID,
+	}
+}
+
+// MigrationTokenArgs holds the arguments required to generate a migration token.
```

Comment:
There's only two args? Maybe just change the signature of NewToken to take a user and tag, not sure why we need a whole args here?